src/context/UserContextProvider.tsx

```diff
@@ -168,10 +170,13 @@ export const UserContextProvider: React.FC = (props): JSX.Element => {
         handleLogin: (data): void =>
           dispatch({ type: ActionType.LOGINUSER, payload: data }),
         // Using useCallback here as these are used in useDataAPI dependency array
-        handleLogout: useCallback(
-          () => dispatch({ type: ActionType.LOGOFFUSER, payload: null }),
-          []
-        ),
+        handleLogout: useCallback(() => {
+          unauthorizedApi().logout({
+            token: localStorage.token,
+          });
+
+          dispatch({ type: ActionType.LOGOFFUSER, payload: null });
```

Comment:
Would it not make more sense to dispatch log out in a unauthorizerdApi().logout callback?

Reply:
Not sure maybe it should stay here to conform to the pattern of the `handleLogin`, `handleRole`, and `handleNewToken` functions

Reply:
Ok. I guess maybe it is even better user experience not to wait.